### PR TITLE
test(bridge-withdrawer): add submitter tests

### PR DESCRIPTION
**Pull request by itamarreif**
## Summary
This adds tests for the bridge withdrawer's `Submitter`. The original intention was to implement these as black box tests, but due to various complications with websockets and eth contracts, decided to hold of on adding those and implemented these instead.

## Background
Sequencer transaction submission logic needed to be tested.

## Changes
- test helpers for the submitter
- sanity check that batches are submitted properly
- tests to verify that failures during submissions causes the submitter to halt

## Testing
- [x] helpers
    - [x] setup
    - [x] response mocks
        - [x] genesis
        - [x] get_nonce
        - [x] broadcast_tx_commit
    - [x] failure mocks
        - [x] get_nonce
        - [x] broadcast_tx_commit
- [x] tests
    - [x] submit sucess
    - [x] submit check_tx failure
    - [x] submit deliver_tx failure

### CODE REVIEW

**Review by aajimal — APPROVED** [review]

**Comment by SuperFluffy** [comment]
From what I can see these are not running the bridge as a blackbox, at least in the Rust put-it-under-`tests/`-sense.

Can you move the tests from inside the crate to outside, as for the other services?

File: crates/astria-bridge-withdrawer/src/withdrawer/submitter/tests.rs [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+}
+
+/// Convert a `Request` object to a `SignedTransaction`
+fn signed_tx_from_request(request: &Request) -> SignedTransaction {
+    use astria_core::generated::protocol::transaction::v1alpha1::SignedTransaction as RawSignedTransaction;
+    use prost::Message as _;
+
+    let wrapped_tx_sync_req: request::Wrapper<tx_sync::Request> =
+        serde_json::from_slice(&request.body)
+            .expect("can't deserialize to JSONRPC wrapped tx_sync::Request");
```
Comment (noot) [inline crates/astria-bridge-withdrawer/src/withdrawer/submitter/tests.rs:249]:
```suggestion
            .expect("able to deserialize to JSONRPC wrapped tx_sync::Request");
```

`expect` language is the opposite of an error message, you want to put what you expected to happen (but didn't, if the expect fails)

File: crates/astria-bridge-withdrawer/src/withdrawer/submitter/tests.rs [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    let raw_signed_tx = RawSignedTransaction::decode(&*wrapped_tx_sync_req.params().tx)
+        .expect("can't deserialize signed sequencer tx from broadcast jsonrpc request");
+    let signed_tx = SignedTransaction::try_from_raw(raw_signed_tx)
+        .expect("can't convert raw signed tx to checked signed tx");
+    debug!(?signed_tx, "sequencer mock received signed transaction");
+
+    signed_tx
+}
```
Comment (noot) [inline crates/astria-bridge-withdrawer/src/withdrawer/submitter/tests.rs:257]:
```suggestion
}

```

nit >.<

**Review by noot — APPROVED** [review]
nice, these look good to me!

**Pull request merged**